src/cache/local.ml

```diff
@@ -31,7 +31,44 @@ module Trimming_result = struct
 end
 
 let default_root () =
-  Path.L.relative (Path.of_string Xdg.cache_dir) [ "dune"; "db"; "v2" ]
+  Path.L.relative (Path.of_string Xdg.cache_dir) [ "dune"; "db" ]
+
+let file_store_version = "v2"
+
+let metadata_store_version = "v2"
+
+let file_store_root cache =
+  Path.L.relative cache.root [ "files"; file_store_version ]
+
+let metadata_store_root cache =
+  Path.L.relative cache.root [ "meta"; metadata_store_version ]
+
+let detect_unexpected_dirs_under_cache_root cache =
+  let expected_in_root path =
+    (* We only report unexpected directories, since quite a few temporary files
+       are created at the cache root, and it would be tedious to keep track of
+       all of them. *)
+    match
+      (Sys.is_directory (Path.to_string (Path.relative cache.root path)), path)
```

Comment:
You can use `Path.is_directory` here so it's a bit cleaner.

Reply:
Indeed, done!